llvm/lib/Target/RISCV/RISCVISelLowering.cpp

```diff
@@ -19517,6 +19517,11 @@ RISCVTargetLowering::shouldExpandAtomicRMWInIR(AtomicRMWInst *AI) const {
   unsigned Size = AI->getType()->getPrimitiveSizeInBits();
   if (Size == 8 || Size == 16)
     return AtomicExpansionKind::MaskedIntrinsic;
```

Comment:
It would be good to use amocas for i8/i16 if both zacas and zabha are supported.